### DISCUSSION TITLE
skeema ignoring test db as system table

### DISCUSSION THREAD

**Discussion by brirams** [skeema/skeema#241] [Ideas and Feedback]
hi all! just getting started with skeema and it's been great so far!

One thing that caught me by surprise when I was setting this up with `skeema init` is that my db named `test` was ignored. It's well documented that system tables like `information_schema`, `performance_schema`, `mysql` and `sys` are excluded but never saw mention of `test` until I dug through the code([ref](https://github.com/skeema/skeema/blob/9de121caa153dfa816be2853e621fe4c7229188d/internal/tengo/instance.go#L433-L437)). 

I wonder if it'd be worthwhile to either:
- update docs to list out what's being ignored by default
- log a warning that a non-system table is being ignored

anyway, not a big deal, as i know to `skeema init --schema test` to get this imported but wanted to bring it up.

Thanks!

**Comment by evanelias** [comment]
Good feedback, thanks! It's actually documented already in a few random places – [schema option doc](https://www.skeema.io/docs/options/#schema), [ignore regex doc](https://www.skeema.io/docs/features/ignoreregex/#ignoring-entire-schemas), [sharding doc](https://www.skeema.io/docs/features/sharding/#mapping-a-directory-to-multiple-schemas-per-server) – but I agree that isn't easily discoverable. It will be added to the `skeema init --help` docs soon, since that is a more intuitive location.

As for why it is ignored: historically, a fresh install of MySQL always came with an empty `test` database out-of-the-box. So Skeema has ignored `test` by default since it was often present, but empty; or it was full of scratch tables that shouldn't be tracked in a schema repo.

In recent years though, that initial `test` database is typically removed upon installation for security reasons by package managers, container entrypoints, etc which either run [`mysql_secure_installation`](https://dev.mysql.com/doc/refman/8.0/en/mysql-secure-installation.html) or just an [explicit DROP](https://github.com/docker-library/mysql/blob/7a02613b637f3b0fa56075663f5de8b3fba3dbee/8.0/docker-entrypoint.sh#L307). I'm not actually sure if first-party builds of recent MySQL or MariaDB versions still create `test` upon server initialization in the first place. If it turns out they don't and haven't for a while, then we can remove its auto-exclusion from Skeema in the near future.

Longer-term, I suspect the core problem here is that `skeema init` defaults to dumping all schemas, which probably was the wrong default, since many users only have 1 relevant schema per database server anyway. Or sometimes they have several but they actually represent different environments so only 1 should be dumped anyway.

So this is something that might change in a future Skeema v2.x, along with a few other unrelated non-backwards-compatible default changes. But that's a bit of a tangent, I'll start a separate RFC discussion thread about those proposals in January 😄

**Comment by evanelias** [comment]
The list of auto-ignored schema names is now shown on `skeema init --help` as of Skeema v1.12.3, released today. And likewise for the [`skeema init` web docs](https://www.skeema.io/docs/commands/init/). Thanks again for the suggestion!
